packages/core/src/SessionEventEmitter.ts

```diff
@@ -0,0 +1,256 @@
+  /**
+   * Register a listener called on successful login.
+   * @param eventName The login event name
+   * @param listener The callback called on successful login
+   */
+  on(eventName: typeof EVENTS.LOGIN, listener: () => void): this;
+  /**
+   * Register a listener called on successful logout.
+   * @param eventName The logout event name.
+   * @param listener The callback called on successful logout.
+   */
+  on(eventName: typeof EVENTS.LOGOUT, listener: () => void): this;
+  /**
+   * Register a listener called on session expiration.
+   * @param eventName The session expiration event name.
+   * @param listener The callback called on session expiration.
+   */
+  on(eventName: typeof EVENTS.SESSION_EXPIRED, listener: () => void): this;
+  /**
+   * Register a listener called on session restoration after a silent login.
+   * @param eventName The session restoration event name.
+   * @param listener The callback called on successful session restore.
+   */
+  on(
+    eventName: typeof EVENTS.SESSION_RESTORED,
+    listener: (currentUrl: string) => unknown
+  ): this;
+  /**
+   * Register a listener called on error, with an error identifier and description.
+   * @param eventName The error event name.
+   * @param listener The callback called on error.
+   */
+  on(
+    eventName: typeof EVENTS.ERROR,
+    listener: (
+      error: string | null,
+      errorDescription?: string | null
+    ) => unknown
+  ): this;
+  /**
+   * Register a listener called on session extension.
+   * @param eventName The session extension event name.
+   * @param listener The callback called on session extension.
+   */
+  on(
+    eventName: typeof EVENTS.SESSION_EXTENDED,
+    listener: (expiresIn: number) => void
+  ): this;
+  /**
+   * Register a listener called when a timeout is set for a session event with
+   * the timeout handle.
+   * @param eventName The timeout set event name.
+   * @param listener The callback called when setting a timeout.
+   */
+  on(
+    eventName: typeof EVENTS.TIMEOUT_SET,
+    listener: (timeoutHandle: number) => void
+  ): this;
+  /**
+   * Register a listener called when a new refresh token is issued for the session.
+   * @param eventName The new refresh token issued event name.
+   * @param listener The callback called when a new refresh token is issued.
+   */
+  on(
+    eventName: typeof EVENTS.NEW_REFRESH_TOKEN,
+    listener: (newToken: string) => void
+  ): this;
+  on(
+    eventName: Parameters<InstanceType<typeof EventEmitter>["on"]>[0],
+    listener: Parameters<InstanceType<typeof EventEmitter>["on"]>[1]
+  ): this {
+    return super.on(eventName, listener);
+  }
+
+  /**
+   * Register a listener called on the next successful login with the logged in WebID.
+   * @param eventName The login event name.
+   * @param listener The callback called on the next successful login.
+   */
+  once(eventName: typeof EVENTS.LOGIN, listener: (webid: string) => void): this;
```

Comment:
This differs from the type signature of the `on` method for the login event which is just `() => void`.

To avoid these kinds of errors can I suggest defining the params as a type; e.g. `type LOGIN_ARGS = [typeof EVENTS.LOGIN, listener: (webid: string) => void)]` so each method could share the type and be consistent.

Reply:
Fixed with b054d7e5